Test/Int/Let.expected

```diff
@@ -3,48 +3,52 @@ f y = 10
 
 query:
 (declare-fun f (Int) Int)
-(assert (not (let ((y 10)) (= (f y) 10))))
 (assert (= (f 10) 10))
+(assert (not (let ((y 10)) (= (f y) 10))))
 (check-sat)
 
 result: unsat
 goal: f 10 = 10
 
 query:
 (declare-fun f (Int) Int)
-(assert (not (= (f 10) 10)))
 (assert (let ((y 10)) (= (f y) 10)))
+(assert (not (= (f 10) 10)))
 (check-sat)
 
 result: unsat
 goal: f z = z
 
 query:
 (declare-fun f (Int) Int)
+(assert (= (f 10) 10))
 (define-fun z () Int 10)
 (assert (not (= (f z) z)))
-(assert (= (f 10) 10))
 (check-sat)
 
 result: unsat
+failed to reconstruct proof: invalid 'simp', proposition expected
```

Comment:
Bug in reconstruction?

Reply:
Probably... This PR has already outgrown its purpose. I will look into this after merging the PR (Tomaz needs this PR to find and fix elaboration issues).

Reply:
Ah gotcha! Okay, let's remember the issues then but merge this now so that Tomaz can make progress.